ToDoApp/ToDoApp/Respository/Respository.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  Respository.swift
+//  ToDoApp
+//
+//  Created by Trong Hieu Vuong on 2022-12-22.
+//
+
+import Foundation
+import FirebaseDatabase
+import FirebaseDatabaseSwift
+protocol Repository {
+    func updateAndCreate<T>(_ value: T) async -> Error? where T: Identifiable, T: Encodable, T.ID == UUID
+    func read<T>() async -> Result<[T], Error> where T: Decodable
+}
+extension DatabaseReference: Repository {
+  // MARK: create a task
+    func updateAndCreate<T>(_ value: T) async -> Error? where T: Identifiable, T: Encodable, T.ID == UUID {
+        do {
+           try await self.child(value.id.uuidString).setValue(value.toDictionary)
+        } catch {
+            return TError.unsavedT
+        }
+        return nil
+    }
+    func read<T>() async -> Result<[T], Error> where T : Decodable {
+        var task: [T] = [T]()
```

Comment:
task array will always be empty since observe is an asynchronous interface.
anyway here getData() method should be used. let me know if you will need help on it

Reply:
Instead of `self.observe`
`try await self.getData` should be used